tests/framework/session/test_session_extension_hooks.py

```diff
@@ -523,21 +523,20 @@ def test_broken_input_update_parallel(
             mock_session_with_broken_before_node_run_hooks.run(runner=ParallelRunner())
 
 
+def wait_and_identity(*args: Any):
```

Comment:
This is a nicer implementation of this fixture ✅.

Out of curiosity, could you explain what you meant in the commit message 'move the inner function out to make the node serializable'?

Reply:
@jmholzer Sure. The original fixture defined an inner function, such object is not picklable and does not work well with multiprocessing. This wasn't an issue until node is added as an argument for hooks. You can simply move the function inside and run the test.

IIRC it's the TestAsyncDataSet.

Reply:
You are more than welcome to test it manually 😁

Reply:
> @jmholzer Sure. The original fixture defined an inner function, such object is not picklable and does not work well with multiprocessing. This wasn't an issue until node is added as an argument for hooks. You can simply move the function inside and run the test.
> 
> IIRC it's the TestAsyncDataSet.

TIL, thanks 😃

> You are more than welcome to test it manually 😁

I tried it out, looks good ✅.